specs/DevSpecs/telemetry.md

```diff
@@ -0,0 +1,187 @@
+# Input.Rating Telemetry Design Specification 
+ 
+## Abstract 
+This design specification covers telemetry integration into the `Input.Rating` control, which will be included in a future release of Adaptive Cards. Telemetry integration is currently slated only for the JavaScript platform; however, this developer specification will be general enough to cover all platforms supported by Adaptive Cards. This telemetry design will focus on sending telemetry via the 1DS platform, but the design is flexible enough to allow for possible integration of other telemetry platforms in the future. See more at the GitHub issue [3058](https://github.com/microsoft/AdaptiveCards/issues/3058).
+ 
+## 1. Feature Dependencies 
+### 1.1 Dependencies on other features 
+Because telemetry will be baked into the `Input.Rating` control as a result of the implementation of this design, this feature is inherently dependent on the `Input.Rating` control. Externally, this feature is dependent on the functionality of the 1DS SDK. 
+ 
+### 1.2 Other features' dependencies on this feature 
+There are currently no other existing features that directly depend on this feature. However, future telemetry integration for other elements will be based on the design presented here. 
+ 
+## 2. Platform-Specific Considerations 
+UWP, JS, and .NET are all currently supported by 1DS, allowing for relatively seamless integration of telemetry for these platforms. Mobile platforms (ie., Android and iOS) are not fully supported by 1DS, so future communication and cooperation with the 1DS team to achieve support for these platforms is necessary. 
+ 
+## 3. Performance Implications 
+> “Perf-penalty in terms of runtime perf on foreground threads should not be affected; all our log event API calls are async. Background threads would kick in when there's time to upload. Total CPU time to upload would depend how much data you logged. In UTC mode - the perf penalty vs. trace logging would be near-zero. About the same.” - Max Golovanov (1DS) 
+
+We will look further into how 1DS affects startup time and memory usage with comparison tests. The real-world perf numbers would depend on telemetry volume per minute. The performance implications shouldn’t be major, but will exist.  
+ 
+ 
+## 4. Backwards Compatibility Concerns  
+If the renderer is not up to date, we will not be able to receive telemetry, which is not a problem because ‘Input.Rating’ would not work correctly without the up-to-date renderer.  
```

Comment:
>ut.Rating’ would not work correctly without the up-to-date renderer.   [](start = 112, length = 70)

Hmm that's not quite accurate. Input.Rating is expected to have a fallback story in place if one is not explicitly specified by the author.. But specifically the requirement here is to ensure that if the renderer with telemetry enabled is not able to be constructed on older hosts, that failure should not result in breaking the functionality of the controls.